File: LeetCode/409/409.py
import collections as co
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = co.Counter(s)
        ans = 0
        for c in count.values():
            ans += c // 2 * 2
            if ans % 2 == 0 and c % 2 == 1:    ans += 1
        print(ans)
        return ans
s = Solution()
s.longestPalindrome('abccccdd')

File: 字符串/例题/LC409-最长回文串/409.cpp
#include <iostream>
using namespace std;
class Solution {
public:
    int longestPalindrome(string s) {
        int ans = 0, count[58] = {0};
        for (auto c : s)
            count[c-65]++;
        for (auto i : count)
            ans += (i-i%2);
        return ans + (ans != s.size());
    }
};
int main(int argc, char const *argv[])
{
    string s = "abccccdd";
    Solution a;
    cout << a.longestPalindrome(s);
    return 0;
}
